Os/CMakeLists.txt

```diff
@@ -97,15 +97,13 @@ register_fprime_module(Os_Test_File_SyntheticFileSystem)
 
 # Basic source files used in every OSAL layer.  Contains common code and helpers.
 set(SOURCE_FILES
-    "${CMAKE_CURRENT_LIST_DIR}/IntervalTimerCommon.cpp"
-    "${CMAKE_CURRENT_LIST_DIR}/Posix/IntervalTimer.cpp"
-
     "${CMAKE_CURRENT_LIST_DIR}/ValidateFileCommon.cpp"
     "${CMAKE_CURRENT_LIST_DIR}/ValidatedFile.cpp"
+    "${CMAKE_CURRENT_LIST_DIR}/IntervalTimer.cpp"
     "${CMAKE_CURRENT_LIST_DIR}/Os.cpp"
+    "${CMAKE_CURRENT_LIST_DIR}/Os.fpp"
```

Comment:
@LeStarch I was not able to find a way to define the `Os.RawTime` abstract type under `Os/Models/` because that causes incorrect include paths in the autocoded files:

```
/Svc/Cycle/CyclePortAc.hpp:19:10: fatal error: Os/Models/RawTime.hpp: No such file or directory
   19 | #include "Os/Models/RawTime.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
```

See error here: https://github.com/nasa/fprime/actions/runs/11300300473/job/31432904439?pr=2923

I was not able to find a way to circumvent or to tell FPP where to find the definition of the RawType (could that make for a feature request?)
I put it in a new `Os/Types.fpp` ... let me know if there's a better approach.

Reply:
I believe that you can use the `at <path>` syntax to point to a header of a different location, but I am ok with what has been done (`Os/Types.fpp`)